docs/source/query-syntax.rst

```diff
@@ -0,0 +1,135 @@
+.. _query-syntax:
+
+Search
+======
+
+MLflow UI and API supports searching runs within a single experiment or a group of experiments
+using a search filter API. This API is simplified version of WHERE clause of SQL syntax.
+
+.. contents:: Table of Contents
+  :local:
+  :depth: 2
+
+Syntax Description
+------------------
+
+Search filter can be a single expression or several expressions separated by ``AND`` keyword.
+Currently, the syntax does not support ``OR``. Each expression has 3 parts: an identifier on
+the LHS, a comparator, and constant on the RHS.
+
+Identifier
+^^^^^^^^^^
+
+Left hand side of a search expressions, signifies a logged entity to compare against. It has two
+sub-parts, a qualifier for and entity you are searching and the name of the entity, separated by
+a period. The qualifier or type of entity you can search can be one of `metrics`, `params`, or
+`tags`.
+
+Example: ``metrics.accuracy``
+
+Constant
+^^^^^^^^
+
+Search syntax requires the RHS of the expression to be a constant. Type of expected constant on
+depends on LHS.
+ - If LHS of the expression is a metric, RHS is expected to be a number (integer or float)
+ - If LHS is a parameter or tag, RHS is expected to be a string constant
+
+Comparator
+^^^^^^^^^^
+
+Two class of comparators are supported.
+ - For string values (`params` and `tags`) only ``=`` and ``!=`` are supported at this time.
+ - Numeric values (for `metrics`) support ``=``, ``!=``, ``>``, ``>=``, ``<``, and ``<=``.
+
+
+Examples
+--------
+
+Subset of runs with logged accuracy metric greater than 0.92.
+
+.. code-block:: sql
+
+  metrics.accuracy > 0.92
+
+
+Runs created using Logistic Regression model and a learning rate (lambda) of 0.001 and
+recorded error metric under 0.05.
+
+.. code-block:: sql
+
+  params.model = "LogisticRegression" and params.lambda = "0.001" and metrics.error <= 0.05
+
+
+Special Cases
+-------------
+
+Entity name containing special characters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+When metric, parameter, or tag names contain special characters like hyphen, space, period, etc
+enclose the entity name within quotes.
```

Comment:
Can we specify that these must be double quotes? Single quotes do not work.

Reply:
I'm editing the file and can make this fix.

Reply:
Good catch. Maybe we should support both types of quotes.